src/main/java/leafout/backend/service/ParkService.java

```diff
@@ -0,0 +1,46 @@
+package leafout.backend.service;
+
+import leafout.backend.model.Exception.ParkException;
+import leafout.backend.model.Park;
+
+
+import java.util.List;
+import java.util.UUID;
+
+public interface ParkService <P extends Park> {
```

Comment:
If `Park` does not have classes inheriting from it, does it make sense to create a generic service for `Park` children? Maybe you want to do it for children of the `Pay` class. Anyway, in my opinion the inheritance structure that you created for Pay, Park and Activity is quite confusing or at least difficult to reason about. Remember you should always favor composition over inheritance, unless you have strong reasons for using inheritance.